e2e-tests/tests/util.ts

```diff
@@ -41,17 +44,33 @@ export async function customRequest(web3: Web3, method: string, params: any[]) {
 	});
 }
 
-export function describeWithExistingNode(title: string, cb: (context: { web3: Web3 }) => void, provider?: string) {
+export function describeWithExistingNode(
+	title: string,
+	cb: (context: { web3: Web3; polkadot: ApiPromise }) => void,
+	providerNodeUrl?: string
+) {
 	describe(title, () => {
 		let context: {
 			web3: Web3;
 			ethersjs: ethers.JsonRpcProvider;
-		} = { web3: null, ethersjs: null };
-
-		if (!provider) {
-			context.web3 = new Web3(LOCAL_NODE_URL);
-		}
-
+			polkadot: ApiPromise;
+		} = {
+			web3: null,
+			ethersjs: null,
+			polkadot: undefined,
+		};
+
+		before(async () => {
+			if (providerNodeUrl) {
+				context.web3 = new Web3(providerNodeUrl);
+				const wsProvider = new HttpProvider(providerNodeUrl);
+				context.polkadot = await new ApiPromise({ provider: wsProvider }).isReady;
```

Comment:
You can remove this line and move line 71 after the `else` block

Reply:
In that case, I would need to create an empty `let wsProvider` variable before the `if` statement, which will then be assigned a value depending on the condition executed. I think it's clearer to have the code executed together, with the minor drawback of having a duplicated line.